### DISCUSSION TITLE
Bobtail: Improved Blockchain Security with Low-Variance Mining

### DISCUSSION THREAD

**Discussion by sa8** [consensus-shipyard/consensuslab#152] [Reading group]
[Bobtail](https://www.ndss-symposium.org/ndss-paper/bobtail-improved-blockchain-security-with-low-variance-mining/) is a proof-of-work system except that instead of having one block whose hash is below the target we have a set of k blocks whose hashes' average is below the target.
The protocol roughly works as follow:
- Assemble transaction to create a block
- Do PoW to find a hash that is "low enough" (according to some probabilities)
- Broadcast the block
- Maintain queue of lowest k blocks (according to their hash value)
- Assemble k blocks whose average hash is below the target  into a superblock
  - all the blocks in a superblock must have the same parent
  - the miner who mined the block with the smallest hash in the set gets to assemble the superblock and propagate it
- Each miner in the set wins a reward
- In the case of conflicting superblocks, choose the superblock with the lowest hash average.

This greatly reduces the chance of double spending as it takes much longer as is much harder for an adversary to create a private superblock.

**Comment by guy-goren** [comment]
Essentially, the main point is to reduce the variance of the block "discovery" distribution. Since minority private chain attacks are typically based on the occurrence of an event different from expectation (e.g., a minority mining two consecutive blocks), reducing the "discovery" variance increases the probability of being close to the expectation (ratio of blocks in the short term is similar to power ratio) and reduces the probability of events further from expectation (a minority getting a short term lead that can be exploited). 

Simply putting (and not very accurate), the system behaves closer to expectation than a single block system, hence, less undesired "surprises".
